packages/adblocker/src/engine/engine.ts

```diff
@@ -1035,6 +1195,8 @@ export default class FilterEngine extends EventEmitter<
 
     // Emit events if we found a match
     if (result.exception !== undefined) {
+      this.emit('filter-matched', result.exception, context);
```

Comment:
instead of those 3 places cannot we have one at the end?
```
if (result.exception || result.filter) {
  this.emit('filter-matched', result.exception || result.filter, {
    request,
    matchType: FilterType.NETWORK,
  });
}
return result;
```